hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -67,8 +68,9 @@ public static Option<HoodieCommitMetadata> resolveWriteConflictIfAny(
       Option<HoodieInstant> lastCompletedTxnOwnerInstant,
       boolean reloadActiveTimeline,
       Set<String> pendingInstants) throws HoodieWriteConflictException {
-    // Skip to resolve conflict if using non-blocking concurrency control
-    if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl() && !config.isNonBlockingConcurrencyControl()) {
+    // Skip to resolve conflict for non bulk_insert operation if using non-blocking concurrency control
+    WriteOperationType operationType = thisCommitMetadata.orElse(new HoodieCommitMetadata()).getOperationType();
+    if (config.needResolveWriteConflict(operationType)) {
```

Comment:
Just passaround an `Option` should be fine? There is no need to new a `HoodieCommitMetadata()`